execution_engine/src/core/engine_state/mod.rs

```diff
@@ -472,6 +472,53 @@ where
                     .ok_or(Error::FailedToGetWithdrawPurses)?
                     .to_owned();
 
+                // Ensure that sufficient balance exists for all unbond purses that are to be
+                // migrated.
+                {
```

Comment:
Consider extracting what we have in this scope to a separate function. That will make it easier to add a unit test.